core/client/hdfs/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -166,8 +166,19 @@ public void useSameContextWithZookeeper() throws Exception {
    */
   @Test(expected = IllegalArgumentException.class)
   public void cannotInitializeWithoutMasterHostname() throws Exception {
-    URI uri = URI.create(Constants.HEADER + "/tmp/path.txt");
-    org.apache.hadoop.fs.FileSystem.get(uri, getConf());
+    String previousValue = "";
+    if (alluxio.Configuration.isSet(PropertyKey.MASTER_HOSTNAME)) {
+      previousValue = alluxio.Configuration.get(PropertyKey.MASTER_HOSTNAME);
+      alluxio.Configuration.unset(PropertyKey.MASTER_HOSTNAME);
+    }
+    try {
+      URI uri = URI.create(Constants.HEADER + "/tmp/path.txt");
+      org.apache.hadoop.fs.FileSystem.get(uri, getConf());
+    } finally {
+      if (!previousValue.isEmpty()) {
+        alluxio.Configuration.set(PropertyKey.MASTER_HOSTNAME, previousValue);
```

Comment:
Do we really want to restore the previous value? Shouldn't we explicitly set `MASTER_HOSTNAME` to empty with an `@Config` rule?

Reply:
Using the Configuration rule to set it to null. Thanks!